ismatec/driver.py

```diff
@@ -55,39 +57,36 @@ async def __aexit__(self, *args):
         """Provide exit to the context manager."""
         self.hw._stop_event.set()
 
-    ####################################################################
-    # Properties or setters/getters                                    #
-    # one per channel for the ones that have the channel kwarg.        #
-    ####################################################################
-
-    async def get_pump_version(self) -> str:
-        """Return the pump model, firmware version, and pump head type code."""
-        return self.hw.query('1#').strip()
-
-    async def get_serial_protocol_version(self) -> int:
-        """Return serial protocol version."""
-        return int(self.hw.query('1x!'))
-
-    async def set_flowrate(self, channel: int, flowrate):
-        """Set the flowrate of the specified channel."""
+    async def get_version(self) -> str:
+        """Get device version information."""
+        pump_version = self.hw.query('1#').strip().split(' ')
+        protocol_version = int(self.hw.query('1x!'))
+        return {
+            'description': pump_version[0],
+            'software': pump_version[1],
+            'pump head': pump_version[2],
+            'protocol': protocol_version,
+        }
```

Comment:
I figured that people are only going to call this when debugging, so they'd want all version information in one request.

They'd also want it in a dict that explains what the response means.

I could be missing a use case, but that's where I got to.